Makefile

```diff
@@ -1,41 +1,126 @@
+	$(shell \
+	echo "100000 * $(2) / ( $(3) * 1000 ) " | \
+	bc )
+	$(QUIET)$(CXX) -c $(DBG) $(6) $(5) -o $$@ $$< -MMD && \
+	if [ $(2) = $(3) ] ; then \
+		printf "[100%%] \033[01;32mBuilt target %s\033[0;0m\n" $(4); \
+	fi
+endef
+
+define generateDirs
+$(1):
+	$(QUIET)echo Creating dir $$@
+	$(QUIET)$(MKDIR) $$@ $(ERRIGNORE)
+endef
+
+# this will recursively expand the above DIRS into full path name
+SRC_DIRS := $(foreach dir, $(DIRS), $(addprefix $(SRC_DIR)/, $(dir)))
+# this will replace the above srcs/ part of the path with bld/
+OBJ_DIRS := $(foreach dir, $(DIRS), $(addprefix $(BLD_DIR)/, $(dir)))
+
+# this will recursively expand the above DIRS into full path name
+UT_SRC_DIRS := $(foreach dir, $(UT_DIRS), $(addprefix $(SRC_DIR)/, $(dir)))
+# this will replace the above srcs/ part of the path with bld/
+UT_OBJ_DIRS := $(foreach dir, $(UT_DIRS), $(addprefix $(BLD_DIR)/, $(dir)))
+
+# this will glob all cpp files inside the SRC_DIRS
+SRCS := $(foreach dir,$(SRC_DIRS),$(wildcard $(dir)/*.cpp))
+UT_SRCS := $(foreach dir,$(UT_SRC_DIRS),$(wildcard $(dir)/*.cpp))
 
 .PHONY: all
 all: $(NAME)
 
-include $(SRC_DIR)/target/generaterules.mk
+# this assigns the .o file names replacing the srcs part of the $(SRCS)
+# path with bld and the .cpp part of the filename with .o
+OBJS := $(foreach src, $(SRCS), $(subst $(SRC_DIR),$(BLD_DIR),$(basename $(src))).o)
+UT_OBJS := $(foreach src, $(UT_SRCS), $(subst $(SRC_DIR),$(BLD_DIR),$(basename $(src))).o)
+
+# I tweaked this rule that George made to build the .d files also inside the
+# bld folder, from the .o files
+DEPS := $(foreach obj, $(OBJS), $(patsubst %.o, %.d, $(obj)))
+UT_DEPS := $(foreach obj, $(UT_OBJS), $(patsubst %.o, %.d, $(obj)))
+
+# Generate rules to compile .o files
+
+define generateBundle
+$(1):
+	$(foreach x, \
+		$(shell for x in $(shell seq 1 $(words $(1))); \
+			do \
+				echo $$x; \
+			done \
+		), \
+		$(eval \
+			$(call generateRules, \
+				$(word $(x),$(1)), \
+				$(x), \
+				$(words $(1)), \
+				$(2), \
+				$(3), \
+				$(4) \
+			) \
+		) \
+	)
+endef
+
+$(call generateBundle, $(OBJS), OBJS, $(INC_PARAMS), $(CXXFLAGS))
+
+$(call generateBundle, $(UT_OBJS), UT_OBJS, $(INC_PARAMS) -Isrcs/incs/test, \
+	$(filter-out -std=c++98,$(CXXFLAGS)) -std=c++11)
+
+# Generate rules to create the dirs that will contain the .o files
+
+$(foreach targetdir, $(OBJ_DIRS), $(eval $(call generateDirs, $(targetdir))))
+
+$(foreach targetdir, $(UT_OBJ_DIRS), $(eval $(call generateDirs, $(targetdir))))
+
+# Here I just changed the name of the OBJ_DIR since we now have several
+$(NAME):  $(OBJ_DIRS) $(OBJS)
+	$(QUIET)$(CXX) $(CXXFLAGS) $(INC_PARAMS) -o $(NAME) $(OBJS)
+	$(QUIET)printf "[100%%] \033[0;32mLinking CXX executable $@\033[0;0m\n"
 
 # take a look at what does a .d file look like to understand this directive
 -include $(DEPS)
+-include $(UT_DEPS)
 
 .PHONY: unit_test
-unit_test: all
-	$(MAKE) -C srcs/app/test/unit
+unit_test:  $(UT_OBJ_DIRS) $(OBJ_DIRS) | $(OBJS) $(UT_OBJS)
```

Comment:
The unit_test rule relinks because it doesn't depend on the unit test binary.

Possible solution:
```diff
diff --git a/Makefile b/Makefile
index 7dc3027..e6421a5 100644
--- a/Makefile
+++ b/Makefile
@@ -21,6 +21,7 @@ INC_DIR = $(SRC_DIR)/incs
 DIRS = . utils config config/analyser server CGI http http/response \
 	http/response/get http/response/post http/response/delete
 
+NAME_TEST = $(BLD_DIR)/utest_app
 UTEST_DIR = test/unit
 UT_DIRS = $(UTEST_DIR) $(UTEST_DIR)/http $(UTEST_DIR)/utils \
           	$(UTEST_DIR)/config_analyser
@@ -107,7 +108,7 @@ $(foreach targetdir, $(OBJ_DIRS), $(eval $(call generateDirs, $(targetdir))))
 $(foreach targetdir, $(UT_OBJ_DIRS), $(eval $(call generateDirs, $(targetdir))))
 
 # Here I just changed the name of the OBJ_DIR since we now have several
-$(NAME):  $(OBJ_DIRS) $(OBJS)
+$(NAME): $(OBJS)
 	$(QUIET)$(CXX) $(CXXFLAGS) $(INC_PARAMS) -o $(NAME) $(OBJS)
 	$(QUIET)printf "[100%%] \033[0;32mLinking CXX executable $@\033[0;0m\n"
 
@@ -115,13 +116,19 @@ $(NAME):  $(OBJ_DIRS) $(OBJS)
 -include $(DEPS)
 -include $(UT_DEPS)
 
-.PHONY: unit_test
-unit_test:  $(UT_OBJ_DIRS) $(OBJ_DIRS) | $(OBJS) $(UT_OBJS)
+$(OBJS): | $(OBJ_DIRS)
+
+$(UT_OBJS): | $(UT_OBJ_DIRS)
+
+$(NAME_TEST): $(OBJS) $(UT_OBJS)
 	$(QUIET)$(CXX) $(filter-out -std=c++98,$(CXXFLAGS)) -std=c++11 $(INC_PARAMS) \
 	-I$(SRC_DIR)/incs/test -o $(BLD_DIR)/utest_app $(filter-out \
 	$(BLD_DIR)/./main.o, $(OBJS)) $(UT_OBJS) -MMD
 	$(QUIET)printf "[100%%] \033[0;32mLinking CXX executable utest_app\033[0;0m\n"
-	$(BLD_DIR)/utest_app
+
+.PHONY: unit_test
+unit_test: $(NAME_TEST)
+	$(NAME_TEST)
 
 .PHONY: integration_test
 integration_test: all

```

Reply:
done, thank you